src/AbstractHPolygon.jl

```diff
@@ -96,11 +96,11 @@ function vertices_list(P::AbstractHPolygon{N},
         return points
     end
     @inbounds for i in 1:n-1
-        points[i] = intersection(Line(P.constraints[i]),
-                                 Line(P.constraints[i+1]))
+        points[i] = element(intersection(Line(P.constraints[i]),
+                                         Line(P.constraints[i+1])))
     end
-    points[n] = intersection(Line(P.constraints[n]),
-                             Line(P.constraints[1]))
+    points[n] = element(intersection(Line(P.constraints[n]),
+                                     Line(P.constraints[1])))
```

Comment:
... and (later) we can move one step further and add a `convex_hull` that deals with a list of singletons (points), thus not needing these `element(..)`. i think it would make the code a bit cleaner.

Reply:
Hm, we could add an iterator type over Singletons that returns their vectors, say, `Singleton2Vector`, and pass `Singleton2Vector(points)` to `convex_hull`. However, if no convex hull shall be applied, we still need to unwrap them.